client-spark/spark-2/src/main/java/org/apache/spark/shuffle/celeborn/SparkShuffleManager.java

```diff
@@ -118,10 +123,24 @@ public <K, V, C> ShuffleHandle registerShuffle(
     String appId = SparkUtils.appUniqueId(dependency.rdd().context());
     initializeLifecycleManager(appId);
 
+    if (!registeredApps.contains(appId)) {
```

Comment:
Does a SparkContext or ShuffleManager or lifecycleManager have multiple apps?

Reply:
@RexXiong, from my knowledge, there is no case that SparkContext or ShuffleManager or lifecycleManager have multiple apps. Could you give some input or thoughts about this?

Reply:
So why do we need registeredApps and fallbackApps sets? For Spark, I think there should only be a fallback for shuffle.